### ISSUE TITLE
Docs: zero grad for "Wasserstein 2 Minibatch GAN"

### ISSUE THREAD

**Issue by kachayev** [PythonOT/POT#464]
Docs page for "Wasserstein 2 Minibatch GAN with PyTorch" has the following snippet of the code:

```python
for i in range(n_iter):
    # generate noise samples
    xn = torch.randn(size_batch, n_features)

    # ...
    # rest of the computation
    # ...

    loss.backward()
    optimizer.step()
```

Typically, we need to set the gradient to zero between batches using `optimizer.zero_grad()`. Is this the case here, or has it been purposefully omitted?

**Comment by rflamary** [comment]
Very good point, I can honestly say that I probably forgot it . I would welcome a PR without it a,d with corrected steps (maybe using SGD instead of RMSPROP?) if you can get a better model with cosntant time (we have very limited power on circlec for building the documentation)

**Comment by kachayev** [comment]
@rflamary Sure, I'll see what's possible. I don't thin `SGD` would make a difference in this specific example but still curious to check others optimizers as well. Will open PR as soon as ready.
